### DISCUSSION TITLE
How to disconnect a device from Thread network?

### DISCUSSION THREAD

**Discussion by OlivierGre** [openthread/openthread#8304] [Q&A] [answered]
Hi,
I'm working on a Thread commissioning demo where I do the following:

- Commissioning of a Thread device
- Start SRP service to indicate the presence of the device
- Send some UDP commands (from a smartphone) to the device

After some time, the smartphone sends a command to the device indicating that we want to disconnect from Thread network.
The Device unregisters the SRP service and reboots.
During init, it swipes all the persistent data. We then do the same things again.

My question is: Do we lack some commands to inform the OTBR that this device should be disconnected from Thread network?
The device starts disconnected (since its persistent memory was erased) but this is maybe not enough.

Thanks for your advices.

**Comment by jwhui** [comment, marked as answer]
Information about the device will eventually time out. The parent maintains a child timeout for each child. Each SRP service has an associated lease time.

To explicitly remove its information before detaching from the network, the following APIs are useful:

- [otThreadDetachGracefully()](https://openthread.io/reference/group/api-thread-general#otthreaddetachgracefully) removes the child table entry from its parent.
- [otSrpClientRemoveHostAndServices()](https://openthread.io/reference/group/api-srp#otsrpclientremovehostandservices) removes host info and all services from the SRP server.

**Comment by OlivierGre** [comment]
Thank you very much Jonathan. I'm going to add a call to otThreadDetachGracefully() (I already had a call to otSrpClientRemoveHostAndServices()).
